shm/Store.hpp

```diff
@@ -0,0 +1,15 @@
+#include "Cargo.hpp"
+#include "Player.hpp"
+
+enum class Response {
+    done,
+    lack_of_money,
+    lack_of_cargo,
+    lack_of_space
+};
+
+class Store{
+public:
+    Response buy(Cargo* cargo, size_t amount, Player* player);
+    Response sell(Cargo* cargo, size_t amount, Player* player);
```

Comment:
I wonder if we should provide `Cargo* cargo` and `size_t amount` or maybe `std::string name` and `size_t amount` ?

`Cargo` object has it's name and amount inside as protected fields, but we still have `amount` provided here separately as an argument to buy and sell functions. 

You don't have to implement it already - it's just a suggestions and maybe we should discuss it on discord?